GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/Controllers/Account/ChangePasswordController/ChangePasswordControllerTests.cs

```diff
@@ -0,0 +1,187 @@
+            Assert.IsTrue(isExpectedPassword);
+        }
+        
+        [Test]
+        [Description("POST: Password is not updated when new password is missing a lowercase letter")]
+        public async Task POST_Password_Is_Not_Updated_When_New_Password_Missing_Lowercase_Letter()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "ABCDEFG1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "ABCDEFG1");
+
+            var controller = new ControllerBuilder<ChangePasswordNewController>()
+                .WithUserId(user.UserId)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .Build();
+            
+            // Required to mock out the Url object when creating the verification URL
+            controller.AddMockUriHelperNew(new Uri("https://localhost:44371/mockURL").ToString());
+            
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            bool isExpectedPassword = await mockUserRepo.CheckPasswordAsync(user, "password");
+            Assert.IsTrue(isExpectedPassword);
+        }
+        
+        [Test]
+        [Description("POST: Password is not updated when new password is different to the confirmation password")]
+        public async Task POST_Password_Is_Not_Updated_When_New_Password_Different_To_Confirmation_Password()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "Password1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "AnotherPassword1");
+
+            var controller = new ControllerBuilder<ChangePasswordNewController>()
+                .WithUserId(user.UserId)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .Build();
+            
+            // Required to mock out the Url object when creating the verification URL
+            controller.AddMockUriHelperNew(new Uri("https://localhost:44371/mockURL").ToString());
+            
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            bool isExpectedPassword = await mockUserRepo.CheckPasswordAsync(user, "password");
+            Assert.IsTrue(isExpectedPassword);
+        }
+

```

Comment:
We could also have a couple of tests to check:
* Is an audit log saved
* Is an email sent

You an use the new `controllerBuilder.DataRepository` to get newly-created data out of the database (just use it like you would the normal repository)